### PR TITLE
feat: Add OptBase2Bytes type

### PR DESCRIPTION
**Pull request by keelerm84**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,6 +3,7 @@ module github.com/launchdarkly/go-configtypes
 go 1.13
 
 require (
+	github.com/alecthomas/units v0.0.0-20231202071711-9a357b53e9c9 // indirect
```
Comment (cwaldren-ld) [inline go.mod:6]: Wonder why this is labeled as `indirect` even though it is a direct dependency 🤔

File: opt_base_2_bytes.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
+package configtypes
+
+import (
+	"encoding/json"
+
+	"github.com/alecthomas/units"
+
+	"gopkg.in/launchdarkly/go-sdk-common.v2/ldvalue"
```
Comment (cwaldren-ld) [inline opt_base_2_bytes.go:8]: Oh wow, this is the really old version of go-sdk-common. I think this should probably be updated to the current package, `github.com/launchdarkly/go-sdk-common/v3`

**Review by cwaldren-ld — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (keelerm84) [reply go.mod]: Another go mod tidy seemed to fix that.

File: opt_base_2_bytes.go (hunk 2 above)
Comment (keelerm84) [reply opt_base_2_bytes.go:8]: I can do that in another PR.

**Pull request merged**